packages/react-vapor/src/components/flatSelect/tests/FlatSelectSelectors.spec.tsx

```diff
@@ -0,0 +1,39 @@
+import { flatSelectInitialState, IFlatSelectState } from '../FlatSelectReducers';
+import { FlatSelectSelectors } from '../FlatSelectSelectors';
+
+describe('FlatSelectSelectors', () => {
+    describe('getInput', () => {
+        it('Should return the default flatSelect state', () => {
+            const FlatSelect = FlatSelectSelectors.getInput({ flatSelect: [] }, { id: 'nothing is selected by the user' });
+            expect(FlatSelect).toEqual(flatSelectInitialState);
+        });
+
+        it('should return the selected FlatSelect', () => {
+            const expectedSelectedFlatSelect: IFlatSelectState = {
+                id: 'the-id',
+                selectedOptionId: 'the-selected-option-id',
+            };
+            const selectedFlatSelect = FlatSelectSelectors.getInput({ flatSelect: [expectedSelectedFlatSelect] }, { id: expectedSelectedFlatSelect.id });
+            expect(selectedFlatSelect).toEqual(expectedSelectedFlatSelect);
+        });
+    });
+});
+
+describe('getSelectedOptionId', () => {
+    it('should return the value of the flatSelect initial state if no other value is chosen', () => {
+        flatSelectInitialState.id = 'the-first-id';

```

Comment:
Here you modify the object which can be quite mind boggling when we try to debug a test (in react-vapor the unit tests are run in a random order)

Do you really need to change the default id and selectedOptionId?